src/app/data/devicesActions.js

```diff
@@ -47,8 +47,9 @@ const addNewDevice = async data => {
 }
 
 const editDeviceById = async data => {
-  const imagesUrl = await uploadDeviceImages(data.images);
-  data.images = [...imagesUrl];
+  const imagesUrl = await uploadDeviceImages(data.newImages);
+  data.images = [...data.images, ...imagesUrl];
+  delete data.newImages;
```

Comment:
Will this direct assignment affect `data.images` from other places?

Reply:
I can see that this will in fact affect the data.images array in case the action fails. So I will patch this by using the spread operator to copy the content of object data into a new variable to use.